### PR TITLE
feat: Chat List All Screen UI

### PR DESCRIPTION
**Pull request by ice-kreios**
### What does this PR do?
This PR implements the UI for the Recent Chats screen. It supports all message types (text, photo, video, document, voice, etc.) 

Additional information

Simulator Screenshot - iPhone 11 - 2024-10-18 at 04 49 17
Simulator Screenshot - iPhone 11 - 2024-10-18 at 04 49 49
Simulator Screenshot - iPhone 11 - 2024-10-18 at 04 55 32

### CODE REVIEW

File: lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+            padding: EdgeInsets.only(right: 10.0.s),
+            child: Assets.svg.iconBlockCheckboxOff.icon(size: 24.0.s),
+          ),
+        //TODO: Animation for scale
+        Flexible(
+          child: AnimatedContainer(
+            curve: Curves.easeInOut,
+            duration: const Duration(milliseconds: 200),
+            child: Row(
+              children: <Widget>[
```
Comment (ice-alcides) [inline lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart:34]: remove please `<Widget>`

File: lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  String _getMessageContentText(RecentChatMessage message, BuildContext context) {
+    switch (message) {
```
Comment (ice-alcides) [inline lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart:144]: Could you please change it to `switch-expression` for better readability?

File: lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+      return Padding(
+        padding: EdgeInsets.only(right: 2.0.s),
+        child: messageIconPath.icon(size: 16.0.s),
+      );
+    }
+    return const SizedBox.shrink();
+  }
+
+  String? _getMessageIcon() {
+    switch (message) {
```
Comment (ice-alcides) [inline lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart:187]: and here

File: lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart (hunk 1 above)
Comment (ice-kreios) [reply lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart]: I removed it. I found out which VSCode extension was adding it and replaced it with a new one that doesn't.

**Review by ice-alcides — APPROVED** [review]

File: lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart (hunk 2 above)
Comment (ice-kreios) [reply lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart]: done

File: lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart (hunk 3 above)
Comment (ice-kreios) [reply lib/app/features/chat/recent_chats/views/components/recent_chat_tile/recent_chat_tile.dart]: done

**Pull request merged**